srtcore/queue.cpp

```diff
@@ -1137,57 +1355,198 @@ void* CRcvQueue::worker(void* param)
            }
            else
            {
-               LOGC(mglog.Fatal, log << self->CONID() << "CChannel reported ERROR DURING TRANSMISSION - IPE. INTERRUPTING worker anyway.");
+               LOGC(mglog.Fatal, log << self->CONID() << "CChannel reported ERROR DURING TRANSMISSION - INTERRUPTING worker anyway.");
            }
            cst = CONN_REJECT;
            break;
        }
+
        // OTHERWISE: this is an "AGAIN" situation. No data was read, but the process should continue.
 
-
-       // take care of the timing event for all UDT sockets
-       uint64_t currtime_tk;
-       CTimer::rdtsc(currtime_tk);
+Handle_timer_events:
```

Comment:
Yes. The difference is that previously it wasn't a problem that after a failed packet handling the loop continues (which means that it won't do the timely event update) because it was doing a "smoothened busy waiting" anyway and it will do this update with the next iteration. This whole rework has changed these rules - the next iteration will only happen at the time of a next event, so doing `continue` here would skip the required event update even if the time might have come for it, and shifting it to the next iteration might run this event update way out of time.

Therefore it's also checked if it occasionally happened that even though the packet came in, the time for the event has also come, and in this case, after skipping the parts that handle the correct packet further (as per handling failure) it should still perform the event update in this iteration.

It might be a good idea to split it later into smaller parts so that this handling can use its own function and therefore be executed after returning from the packet handling, no matter what status it returned.